boards: shields: add rk043fn02h_ct and rk043fn66hs_ctg shields

danieldegrasse opened the pull request:
Add `rk043fn02h_ct` and `rk043fn66hs_ctg` shields. These shields both describe Rocktech panels available from NXP, and compatible with several NXP iMX RT EVKs using a FPC interface.

Define nodelabel aliases for these FPC interfaces, and update the RT1060 EVK board definition to no longer define the RK043FN02H_CT as a panel at the board level (and use the shield instead)

Support for additional NXP EVKs (RT1064, RT1050, RT1040) will be added in follow up PRs.

danieldegrasse committed:
boards: nxp: mimxrt1060_evk: define parallel and i2c FPC interfaces

Define parallel and i2c FPC display interfaces for the RT1060 EVK. These
interfaces are used by multiple displays for the EVK.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
boards: shields: rk043fn02h_ct: add shield for rk043fn02h_ct

Add a shield for the RK043FN02H-CT panel, a Rocktech display panel with
an FT5336 touch controller. This panel uses a 40+6 FPC interface for
parallel displays, which is supported by many NXP iMX RT EVKs.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
boards: shields: rk043fn66hs_ctg: add shield for rk043fn66hs_ctg

Add a shield for the RK043FN66HS-CTG panel, a Rocktech display panel with
an GT911 touch controller. This panel uses a 40+6 FPC interface for
parallel displays, which is supported by many NXP iMX RT EVKs.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
boards: nxp: mimxrt1060_evk: remove display definition

Remove display definition for the RT1060 EVK. This display panel is now
supported as the RK043FN02H-CT shield. Add this information to the
documentation for the board, along with other tested shields.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
samples: add testcases for rk043fn02h_ct and rk043fn66hs_ctg shields

Add testcases for rk043fn02h_ct and rk043fn66hs_ctg shields to display
and LVGL samples, so these testcases will be built by CI.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

DerekSnell reviewed (changes requested):
@danieldegrasse and @dleach02 ,
This PR works well on my hardware.  I tested with the following combinations:
* `mimxrt1050_evk` board with `rk043fn02h_ct` shield
* `mimxrt1060_evkb` board with `rk043fn66hs_ctg` shield

The doc page has some typos to fix for the build instructions.  Otherwise, this looks good to me.

danieldegrasse commented:
> @danieldegrasse and @dleach02 , This PR works well on my hardware. I tested with the following combinations:
> 
>     * `mimxrt1050_evk` board with `rk043fn02h_ct` shield
> 
>     * `mimxrt1060_evkb` board with `rk043fn66hs_ctg` shield
> 
> 
> The doc page has some typos to fix for the build instructions. Otherwise, this looks good to me.

Thanks for catching these- both should be fixed now

dleach02 commented:
@kartben need your approval

dleach02 commented:
@kartben please give this attention or reassign it to someone else (me)

danieldegrasse committed:
dts: bindings: gpio: add bindings for NXP display interfaces

Add bindings for nxp display interfaces. These bindings describe the 6
and 40 pin FPC connectors used for displays on several NXP EVKs using
gpio nexus nodes.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse commented on boards/nxp/mimxrt1060_evk/mimxrt1060_evk.dts:
```diff
@@ -39,6 +37,33 @@
+		gpio-map-pass-thru = <0 0x3f>;
+		gpio-map =	<1  0 &gpio1 2 0>,	/* Pin 2, LCD touch RST */
+				<2  0 &gpio1 11 0>;	/* Pin 3, LCD touch INT */
+	};
```
Good point. I've added compatibles for both connections

dleach02 commented:
@kartben need you to review Daniel's change.

@MaureenHelm @DerekSnell please review as your approvals were dropped when @danieldegrasse addressed @kartben change request.

MaureenHelm commented on dts/bindings/gpio/nxp-i2c-tsc-fpc.yaml:
```diff
@@ -0,0 +1,20 @@
+# SPDX-License-Identifier: Apache-2.0
+
+
+compatible: "nxp-i2c-tsc-fpc"
```
```suggestion
compatible: "nxp,i2c-tsc-fpc"
```

MaureenHelm commented on dts/bindings/gpio/nxp-parallel-lcd-connector.yaml:
```diff
@@ -0,0 +1,33 @@
+# SPDX-License-Identifier: Apache-2.0
+
+
+compatible: "nxp-parallel-lcd-connector"
```
```suggestion
compatible: "nxp,parallel-lcd-connector"
```

MaureenHelm reviewed (commented):
It seems we're inconsistent about whether the vendor should be followed by a comma or a dash in the gpio header bindings. But it looks strange not to have a comma

danieldegrasse commented:
> It seems we're inconsistent about whether the vendor should be followed by a comma or a dash in the gpio header bindings. But it looks strange not to have a comma

Good point- both bindings have been updated to have a comma
